hilti/src/compiler/codegen/operators.cc

```diff
@@ -307,15 +307,13 @@ struct Visitor : hilti::visitor::PreOrder<std::string, Visitor> {
     result_t operator()(const operator_::map::Get& n) {
         auto [self, args] = methodArguments(n);
 
-        std::string x = args[0];
+        std::string k = args[0];
         ;
 
-        if ( auto default_ = optionalArgument(args, 1); ! default_.empty() ) {
-            x += ", ";
-            x += default_;
-        }
-
-        return fmt("%s.get(%s)", self, x);
+        if ( auto default_ = optionalArgument(args, 1); ! default_.empty() )
+            return fmt("(%s.contains(%s)? %s.get(%s) : %s)", self, k, self, k, default_);
```

Comment:
`k` is evaluated twice here now, which can be problematic in cases where it has a side effect (it can end up being any kind of C++ expression). Not sure what's the best way around that, we can't use a tmp local either because we're inside an expression. It might need a helper function of some kind.

Reply:
Good catch. I will introduce a lambda here which binds `k` once.